commit ce70d151352a459f4971733c98c1bab1d5a32e6a
Author: q66 <q66@chimera-linux.org>
Date:   Sun Nov 24 18:42:17 2024 +0100

    Fix SVCDSCDIR replies on 64-bit big endian architectures
    
    While on LE this works, on BE it copies the other four bytes into
    the uint32_t, resulting in a zero. This in turn results in a
    protocol error on those architectures.

diff --git a/src/control.cc b/src/control.cc
index 7b9aab0..5898543 100644
--- a/src/control.cc
+++ b/src/control.cc
@@ -1315,7 +1315,7 @@ bool control_conn_t::process_query_dsc_dir()
     // 4 bytes (uint32_t) = directory length (no nul terminator)
     // N bytes            = directory (no nul)
     std::vector<char> reppkt;
-    size_t sdir_len = strlen(service->get_service_dsc_dir());
+    auto sdir_len = static_cast<uint32_t>(strlen(service->get_service_dsc_dir()));
     reppkt.resize(1 + sizeof(uint32_t) + sdir_len);  // packet type, dir length, dir
     reppkt[0] = (char)cp_rply::SVCDSCDIR;
     std::memcpy(&reppkt[1], &sdir_len, sizeof(sdir_len));
